controllers/ratelimitpolicy_limits.go

```diff
@@ -105,8 +115,91 @@ func (r *RateLimitPolicyReconciler) buildRateLimitIndex(ctx context.Context, rlp
 			return nil, err
 		}
 
+		r.applyOverrides(ctx, rlp, t)
+
 		rateLimitIndex.Set(rlpKey, rlptools.LimitadorRateLimitsFromRLP(rlp))
 	}
 
 	return rateLimitIndex, nil
 }
+
+// applyOverrides checks for any overrides set for the RateLimitPolicy.
+// It iterates through the slice of policies to find overrides for the provided target HTTPRoute.
+// If an override is found, it updates the limits in the RateLimitPolicySpec accordingly.
+func (r *RateLimitPolicyReconciler) applyOverrides(ctx context.Context, rlp *kuadrantv1beta2.RateLimitPolicy, t *kuadrantgatewayapi.Topology) {
+	logger, _ := logr.FromContext(ctx)
+	logger = logger.WithName("applyOverrides")
+
+	topologyIndexes := kuadrantgatewayapi.NewTopologyIndexes(t)
+
+	var policies []kuadrantgatewayapi.Policy
+	// For each gw, get all the policies if the current rlp is within the topology for this gateway
+	for _, gw := range t.Gateways() {
+		policyList := topologyIndexes.PoliciesFromGateway(gw.Gateway)
+		policyKeys := utils.Map(policyList, func(p kuadrantgatewayapi.Policy) client.ObjectKey {
+			return client.ObjectKeyFromObject(p)
+		})
+
+		// this policy is potentially affected by other policies from this gateway
+		if slices.Contains(policyKeys, client.ObjectKeyFromObject(rlp)) {
+			policies = append(policies, policyList...)
+		}
+	}
+
+	filteredPolicies := utils.Filter(policies, func(policy kuadrantgatewayapi.Policy) bool {
+		// HTTPRoute RLPs should only care about overrides from gateways
+		if kuadrantgatewayapi.IsTargetRefHTTPRoute(rlp.GetTargetRef()) {
```

Comment:
For a gateway `rlp`, this will be false for all `policies`. Therefore, `filterPolicies` will be an empty list. Maybe avoid calling `applyOverrides` altogether when `rlp` is known to be at the top of the hierarchy (i.e. when `rlp` is a gateway policy)?